helper/imexport.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * DokuWiki Plugin struct (Helper Component)
+ *
+ * @license GPL 2 http://www.gnu.org/licenses/gpl-2.0.html
+ * @author  Andreas Gohr, Michael Große <dokuwiki@cosmocode.de>
+ */
+
+// must be run within Dokuwiki
+use dokuwiki\plugin\struct\meta\Schema;
+
+if(!defined('DOKU_INC')) die();
+
+class helper_plugin_struct_imexport extends DokuWiki_Plugin {
+
+    private $sqlite;
+
+
+    /**
+     * this possibly duplicates @see helper_plugin_struct::getSchema()
+     */
+    public function getAllSchemasList() {
+        return Schema::getAll();
+    }
+
+    /**
+     * Delete all existing assignment patterns of a schema and replace them with the provided ones.
+     *
+     * @param string   $schemaName
+     * @param string[] $patterns
+     */
+    public function replaceSchemaAssignmentPatterns($schemaName, $patterns) {
+        /** @var \helper_plugin_struct_db $helper */
+        $helper = plugin_load('helper', 'struct_db');
+        $this->sqlite = $helper->getDB();
+        $schemaName = $this->sqlite->escape_string($schemaName);
+        $sql = array();
+        $sql[] = "DELETE FROM schema_assignments_patterns WHERE tbl = '$schemaName'";
+        $sql[] = "DELETE FROM schema_assignments WHERE tbl = '$schemaName'";
+        foreach ($patterns as $pattern) {
+            $pattern = $this->sqlite->escape_string($pattern);
+            $sql[] = "INSERT INTO schema_assignments_patterns (pattern, tbl) VALUES ('$pattern','$schemaName')";
+        }
+
+        $this->sqlite->doTransaction($sql);
+        $assignments = new \dokuwiki\plugin\struct\meta\Assignments();
+        $assignments->propagatePageAssignments($schemaName);
+    }
+
+    /**
+     * Returns array of patterns for the given Schema
+     *
+     * @param string $schemaName
+     * @return string[]
+     */
+    public function getSchemaAssignmentPatterns($schemaName) {
+        /** @var \helper_plugin_struct_db $helper */
+        $helper = plugin_load('helper', 'struct_db');
+        $this->sqlite = $helper->getDB();
+
+        $sql = 'SELECT pattern FROM schema_assignments_patterns WHERE tbl = ?';
+        $res = $this->sqlite->query($sql, $schemaName);
+        $patterns = $this->sqlite->res2arr($res);
+        $this->sqlite->res_close($res);
+        return array_map(function($elem){return $elem['pattern'];},$patterns);
+    }
+
+    /**
+     * Get the json of the current version of the given schema.
+     * If the schema doesn't exist, then the returned JSON will have id 0.
```

Comment:
Not sure, because a missing schema is in this usecase not an error per-se. However if we'd introduce a `schemaExists($schemaName)` function, then we could treat a non-existing schema here as an error.